File: 20220907/모범답안/20220907_07.js/slice.js
function solution(A){
    var count = 0;
  
    A.sort(function(a, b){
      return (a[1] - a[0]) - (b[1] - b[0]);
    });
  
    while(A.length > 0){
      var temp = A.shift();
      count++;
  
      A.filter(function(a, b){
        if(temp[0] <= a[1] && temp[1] >= a[0]){
          A.splice(b, 1);
        }
      });
    }
    return count;
  }

File: 20220907/모범답안/20220907_07.js/slice.py
def solution(A):
    count = 0

    A = sorted(A, key = lambda x: x[1] - x[0])

    while len(A) > 0:
        temp = A[0]
        A.remove(A[0])
        count += 1

        for a in filter(lambda x : x[0] <= temp[1] and x[1] >= temp[0], A):
            A.remove(a)

    return count